app/code/core/Mage/Catalog/Model/Resource/Abstract.php

```diff
@@ -699,6 +699,45 @@ public function getAttributeRawValue($entityId, $attribute, $store)
         return $attributesData ? $attributesData : false;
     }
 
+    /**
+     * Retrieve attribute's raw text value from DB.
+     *
+     * @param int $entityId
+     * @param int|string|array $attribute atrribute's ids or codes
+     * @param int|Mage_Core_Model_Store $store
+     * @return bool|string|array
+     */
+    public function getAttributeRawText($entityId, $attribute, $store)
+    {
+        if (!$entityId || empty($attribute)) {
+            return false;
+        }
+
+        if ($store instanceof Mage_Core_Model_Store) {
+            $store = $store->getId();
+        }
+
+        $store = (int)$store;
+        $attribute = is_array($attribute) ? $attribute : [$attribute];
+        $value = $this->getAttributeRawValue($entityId, $attribute, $store);
+
+        if (!$value) {
+            return false;
+        }
+
+        // Ensure we have an associative array of attribute => values
+        $values = is_array($value) ? $value : array_combine($attribute, [$value]);
+
+        foreach ($values as $_attribute => &$_value) {
+            $_attribute = (clone $this->getAttribute($_attribute))->setStoreId($store);
+            if ($_attribute->getFrontendInput() === 'select' || $_attribute->getFrontendInput() === 'multiselect') {
```

Comment:
To take care of `boolean` and custom frontend input, e.g., I have `select_country`:
```suggestion
            if ($_attribute->getSourceModel() || $_attribute->getFrontendInput() === 'select' || $_attribute->getFrontendInput() === 'multiselect') {
```